hlx_statics/scripts/lib-adobeio.js

```diff
@@ -193,13 +195,46 @@ export function decorateInlineCodes(element) {
  * @param {*} container The container to inspect
  */
 export function buildCodes(container) {
-  const codes = [...container.querySelectorAll('main > div > pre > code')];
+  const codes = [...container.querySelectorAll('main > div pre')];
+
   codes.forEach((code) => {
+
+    const parentDiv = code.closest('div');
+    parentDiv.classList.add('code-container', 'codeblock-container');
     const block = buildBlock('code', code.outerHTML);
-    block.classList.add('block');
-    const parentContainer = code.parentElement.parentElement;
-    const pre = parentContainer.querySelector('pre');
-    pre.replaceWith(block);
+
+    if (code) {
+      const wrapperDiv = document.createElement('div');
+      const blockDiv = document.createElement('div');
+
+      wrapperDiv.style.margin = "1em 0";
+      code.style.whiteSpace = "pre-wrap";
+
+      code.parentNode.insertBefore(wrapperDiv, code);
+
+      wrapperDiv.classList.add('code-wrapper')
+      blockDiv.classList.add('code', 'block');
+
+      blockDiv.appendChild(code);
+      wrapperDiv.appendChild(blockDiv);
+
+      [decorateBlock, decorate].forEach(fn => fn(blockDiv));
```

Comment:
As discussed on slack, this might feel a bit hacky (because we don't need to explicitly call `decorateBlock` on the other blocks). 

However, @BaskarMitrah had pointed out that it doesn't call the decorateBlock because the code block is placed inside the li of the default wrapper content. So I think we can leave this in, especially since the fix works